broadinstitute/single_cell_portal_core
# Fix Image Pipeline integration (SCP-4841)

This fixes a handful of miscellaneous issues preventing a successful run of Image Pipeline from the Rails console.

It complements changes in #1682 by refining code from #1650 and #1663.  

Testing this code is currently only feasible on staging.  I just manually tested it as described below.  Those steps are likely only worthwhile as a reference, as #1687 does further integration that depends on this to fully work.  Tests in that PR supersede these.

To test:
1. Go to https://scp-jenkins.dsp-techops.broadinstitute.org/job/scp-deploy-hotfix-to-staging/
2. Click "Configure"
3. Ensure "ew-fix-image-pipeline" appears twice, once for "Branch Specifier" and once for "export GIT_BRANCH="
4. Click "Save"
5. Click "Build Now", wait for successful build
6. Tunnel into staging, 
7. Enter running Docker container: `sudo -i` then `docker exec -it single_cell bash`
8. Enter staging Rails console: `bin/rails console --environment staging`
9. Kick off Image Pipeline run via Rails console: 
`study = Study.find_by(name: '0.10.3 1.3M 10x mouse'); cluster_file = study.cluster_ordinations_files.first; ImagePipelineService.run_image_pipeline_job(study, cluster_file)`
10. Wait 3-5 minutes, go to bucket: https://console.cloud.google.com/storage/browser/fc-a99b2449-4549-4595-924b-b8d2f1f36785/_scp_internal/parse_logs
11. Find row with recent timestamp, enter that "directory", open log file
12. Confirm at least one line says "uploaded to destination"

This satisfies SCP-4841.
